server/service/service_osquery.go

```diff
@@ -454,22 +452,19 @@ func (svc *Service) detailQueriesForHost(ctx context.Context, host fleet.Host) (
 	return queries, nil
 }
 
-func (svc *Service) shouldUpdate(lastUpdated time.Time, interval time.Duration) bool {
+func (svc *Service) shouldUpdate(lastUpdated time.Time, interval time.Duration, hostID uint) bool {
 	var jitter time.Duration
 	if svc.config.Osquery.MaxJitterPercent > 0 {
-		maxJitter := time.Duration(svc.config.Osquery.MaxJitterPercent) * interval / time.Duration(100.0)
-		randDuration, err := rand.Int(rand.Reader, big.NewInt(int64(maxJitter)))
-		if err == nil {
-			jitter = time.Duration(randDuration.Int64())
-		}
+		maxJitter := int64(svc.config.Osquery.MaxJitterPercent) * interval.Milliseconds() / 100.0
+		jitter = time.Duration((int64(hostID)+svc.getJitterSeed())%maxJitter) * time.Millisecond
```

Comment:
`(int64(hostID)+svc.getJitterSeed()` could overflow, and `jitter` would be negative.

Reply:
So the overflow could be up to `int64(hostID)` in milliseconds, so I think it would be fine.

Reply:
Nvm, I cannot math xD